metalgpy/_sample.py

```diff
@@ -36,11 +36,11 @@ def list_type(list_exp):
             return LIST_KCATEGORICAL
 
 class BaseSampler:
-    def __init__(self, exp, distributions: List[tuple]=None, rng:np.random.RandomState=None):
+    def __init__(self, exp, distributions, rng:np.random.RandomState=None):
```

Comment:
the `distributions` parameter can `None` by default and `dict` otherwise, `distributions: dict=None`. In the `None` case, we will introspect the `exp` to collect the default `_dist`.